src/components/totals/GoalProgress/index.js

```diff
@@ -1,124 +1,102 @@
 "use strict";
 
-var _         = require('lodash');
-var React     = require('react');
-var campaigns = require('../../../api/campaigns');
-var Icon      = require('../../helpers/Icon');
-var numeral   = require('numeral');
+var React           = require('react/addons');
+var PureRenderMixin = React.addons.PureRenderMixin;
+var I18nMixin       = require('../../mixins/I18n');
+var Icon            = require('../../helpers/Icon');
+var numeral         = require('numeral');
 
 module.exports = React.createClass({
   displayName: "GoalProgress",
+
+  mixins: [PureRenderMixin, I18nMixin],
+
   propTypes: {
-    goal: React.PropTypes.number,
-    campaignUid: React.PropTypes.string.isRequired,
-    format: React.PropTypes.string
+    total: React.PropTypes.number.isRequired,
+    goal: React.PropTypes.number.isRequired,
+    format: React.PropTypes.string,
+    currencySymbol: React.PropTypes.string,
+    i18n: React.PropTypes.object
   },
 
   getDefaultProps: function() {
     return {
-      goal: 0,
-      format: '0,0'
+      format: '0,0',
+      currencySymbol: '$',
+      defaultI18n: {
+        text: '**{total}** raised of **{goal}** goal'
+      }
     };
   },
 
   getInitialState: function() {
     return {
-      isLoading: false,
-      total: 0,
-      currencySymbol: '$'
+      progress: this.getProgress()
     };
   },
 
-  componentWillMount: function() {
-    this.loadCampaignGoal();
-  },
-
-  loadCampaignGoal: function() {
-    this.setState({ isLoading: true });
-    campaigns.find(this.props.campaignUid, this.onSuccess);
+  componentDidUpdate: function() {
+    this.setState({ progress: this.getProgress() });
   },
 
-  onSuccess: function(result) {
-    var campaign = result.campaign;
-    var funds_raised = campaign.funds_raised || {};
-    var total = funds_raised.cents || 0;
-    var currency = funds_raised.currency || {};
-    var currencySymbol = currency.symbol || '$';
-
-    this.setState({
-      isLoading: false,
-      total: total,
-      currencySymbol: currencySymbol
-    });
+  getProgress: function() {
+    var props = this.props;
+    return Math.min(props.total / props.goal, 1);
   },
 
   renderProgressBar: function() {
-    var total = this.state.total;
-    var goal = this.props.goal;
-    var offsetWidth = numeral(total / goal).format('0%');
+    var progress = this.state.progress;
+    var offsetWidth = numeral(progress).format('0%');
     var style = { width: offsetWidth || '100%' };
-    var classes = 'GoalProgress__bar';
 
-    if (offsetWidth == '100%') {
-      classes += '--completed';
-    } else if (offsetWidth == '0%') {
+    if (offsetWidth === '0%') {
       style = { width: '0.001%'}; // Forces zero width cell in Chrome
     }
```

Comment:
Now that you have changed `.GoalProgress__barFill` to not be a table cell I think this could be removed - have you tried that already?